Google.Api.Gax.Grpc/Rest/RestMethod.cs

```diff
@@ -85,4 +85,16 @@ internal async Task<TResponse> ReadResponseAsync<TResponse>(Task<ReadHttpRespons
         }
         return (TResponse) _parser.Parse(httpResponse.Content, _protoMethod.OutputType);
     }
+
+    // TODO: Handle cancellation?
+    /// <summary>
+    /// Converts the http response into a IAsyncStreamReader.
+    /// </summary>
+    internal IAsyncStreamReader<TResponse> ResponseStreamAsync<TResponse>(Task<HttpResponseMessage> httpResponseTask)
+    {
+        var httpResponse = httpResponseTask.ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
This makes me nervous: it's going to block until the task has completed. Why isn't this an async method using await?

Reply:
(Or if we can't make this async, we should move the "open the stream" part into PartialDecodingStreamReader, so that *that* can await it appropriately.)

Reply:
Refactored to address this and push the first wait to the MoveAsync.
Also added some unit tests.